src/usePopper.js

```diff
@@ -0,0 +1,121 @@
+import PopperJS from 'popper.js';
+
+import { useCallback, useEffect, useState, useRef } from 'react';
+
+const initialPopperStyles = {
+  position: 'absolute',
+  top: '0',
+  left: '0',
+  opacity: '0',
+  pointerEvents: 'none',
+};
+
+const initialArrowStyles = {};
+
+const getRefValue = refOrValue =>
+  refOrValue !== null && 'current' in refOrValue
+    ? refOrValue.current
```

Comment:
can we just require this to be `ref.current`, and not try to be cute here? i'm worried about things possibly going wrong. it doesn't seem like this gets us much DX-wise

Reply:
we need to allow the straight ref case, b/c it's essentially required for doing updates in response to a ref being set. I do hate this logic tho, maybe we can go the other way and require only callback refs?

Reply:
What do you mean? The caller of `usePopper` can always pass in an unwrapped ref for the arrow, no?

Reply:
ooo right this is only for the arrow. hmm yeah maybe, tho it's sort of weird to have an api that only takes a `useRef` ref and not any ref? idk actually, it's odd that there is this distinction

Reply:
i'm suggesting we kill `getRefValue` and force the caller to pass in e.g. `arrowElementRef.current`